packages/hosted-dropin-integration/src/HostedDropInPaymentMethodComponent.tsx

```diff
@@ -0,0 +1,403 @@
+import {
+    CardInstrument,
+    CheckoutSelectors,
+    CheckoutService,
+    CustomerInitializeOptions,
+    CustomerRequestOptions,
+    Instrument,
+    LanguageService,
+    PaymentInitializeOptions,
+    PaymentInstrument,
+    PaymentMethod,
+    PaymentRequestOptions,
+} from '@bigcommerce/checkout-sdk';
+import { memoizeOne } from '@bigcommerce/memoize';
+import classNames from 'classnames';
+import { find, noop, some } from 'lodash';
+import React, { Component, ReactNode } from 'react';
+
+import {
+    CardInstrumentFieldset,
+    CreditCardValidation,
+    isBankAccountInstrument,
+    isCardInstrument,
+    isInstrumentCardCodeRequiredSelector,
+    isInstrumentCardNumberRequiredSelector,
+    isInstrumentFeatureAvailable,
+} from '@bigcommerce/checkout/instrument-utils';
+import { PaymentFormService } from '@bigcommerce/checkout/payment-integration-api';
+import { LoadingOverlay } from '@bigcommerce/checkout/ui';
+
+export interface HostedDropInPaymentMethodProps {
+    checkoutService: CheckoutService;
+    checkoutState: CheckoutSelectors;
+    containerId: string;
+    hideVerificationFields?: boolean;
+    hideWidget?: boolean;
+    isInitializing?: boolean;
+    isSignInRequired?: boolean;
+    isUsingMultiShipping?: boolean;
+    language: LanguageService;
+    method: PaymentMethod;
+    paymentForm: PaymentFormService;
+    shouldHideInstrumentExpiryDate?: boolean;
+    deinitializeCustomer?(options: CustomerRequestOptions): Promise<CheckoutSelectors>;
+    deinitializePayment(options: PaymentRequestOptions): Promise<CheckoutSelectors>;
+    initializeCustomer?(options: CustomerInitializeOptions): Promise<CheckoutSelectors>;
+    initializePayment(
+        options: PaymentInitializeOptions,
+        selectedInstrumentId?: string,
+    ): Promise<CheckoutSelectors>;
+    onSignOut?(): void;
+    onSignOutError?(error: Error): void;
+    onUnhandledError?(error: Error): void;
+    signInCustomer?(): void;
+    validateInstrument?(shouldShowNumberField: boolean): React.ReactNode;
+}
+
+interface HostedDropInPaymentMethodState {
+    isAddingNewCard: boolean;
+    selectedInstrumentId?: string;
+}
+
+interface HostedDropInPaymentMethodHelperMethods {
```

Comment:
```suggestion
interface HostedDropInPaymentMethodDerivedProps {
```
?

Reply:
I converted HOC into this. Will rename them in this and another PR.